### PR TITLE
doc: add commands

### PR DESCRIPTION
**Pull request by edouardparis**

### CODE REVIEW

File: doc/API.md [hunk 1]
```diff
@@ -213,3 +213,27 @@ then the revault daemon will start the spending process.
...
+  | <----sign spend tx-+ |                          |
+  | +------sig---------> |                          |
+  +                      |                          |
+                         |                          |
+client 2                 |                          |
+  +                      |                          |
+  | <---sign psbt------+ |                          |
+  | +-----psbt---------> |                          | // daemon ask wt opinion
+  +                      | +-------signtx psbt----> | // if ack, cosign server sign
+                         | <---OK or wt nack------+ | // then daemon broadcasts tx
```
Comment (darosior) [inline doc/API.md:234]: Note that WT opinion has already been polled, and final validation is done *after* requesting the sig from the cosig

**Review by darosior — COMMENTED** [review]
Nice, i disagree on the flow (and slightly niting on the names) but otherwise looks good :)

File: doc/API.md [hunk 2]
```diff
@@ -111,4 +134,28 @@ amount of the output.
...
+| --------- | ------- | ------------------------------------------------------------------------------------------------------------- |
+| tx        | string  | PSBT with the required signatures                                                                             |
+| type      | string  | Type of the transaction, can be `emergency_tx`, `cancel_tx`, `emergency_unvault_tx`, `unvault_tx`, `spend_tx` |
+
+#### Response
+
+| Field    | Type | Description                                                                                             |
+| -------- | ---- | ------------------------------------------------------------------------------------------------------- |
+| sigs_ack | bool | Watchower has stored the given revocation tx (Field not present if tx type is `unvault_tx`, `spend_tx`) |
+
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply doc/API.md:161]: Ok I agree that the call would be more explicit with `spend_tx` etc

File: doc/API.md [hunk 3]
```diff
@@ -78,26 +80,47 @@ filtered by an optional `status` parameter.
 | ------------- | ------------------------------------------ | ------------------------- |
 | `vaults`      | array of [vault resource](#vault-resource) | Vaults filtered by status |
 
-### `signvault`
+### `securevault`
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply doc/API.md:83]: I will change it for `get_revocation_txs` as suggested below

File: doc/API.md [hunk 4]
```diff
@@ -78,26 +80,47 @@ filtered by an optional `status` parameter.
...
 #### Request
 
-| Parameter        | Type    | Description                         |
-| ---------------- | ------- | ----------------------------------- |
-| `txid`           | string  | Unique ID of the vault to sign      |
-| `only_emergency` | boolean | Sign only the emergency transaction |
+| Parameter        | Type    | Description                                     |
+| ---------------- | ------- | ----------------------------------------------- |
+| `txid`           | string  | Unique ID of the vault to secure                |
+| `only_emergency` | boolean | Retrieve only the emergency transaction to sign |
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply doc/API.md:93]: OK to remove it

File: doc/API.md [hunk 5]
```diff
@@ -3,11 +3,16 @@
...
+| Command                                   | Description                                          |
+| ----------------------------------------- | ---------------------------------------------------- |
+| [`getinfo`](#getinfo)                     | Display general information                          |
+| [`listvaults`](#listvaults)               | Display a paginated list of vaults                   |
+| [`getrevocationtxs`](#getrevocationtxs)   | Retrieve the Revault revocation transactions to sign |
+| [`getunvaulttx`](#getunvaulttx)           | Retrieve the Revault unvault transaction to sign     |
+| [`getspendtx`](#getspendtx)               | Retrieve the Revault spend transaction to sign       |
+| [`revocationtxs`](#revocationtxs)         | Give back the revocation transactions signed         |
+| [`unvaulttx`](#unvaulttx)                 | Give back the unvault transaction signed             |
+| [`spendtx`](#spendtx)                     | Give back the spend transaction signed               |
```
Comment (darosior) [inline doc/API.md:15]: It looks like a *nice* interface to me :+1:

File: doc/API.md [hunk 6]
```diff
@@ -78,26 +80,47 @@ filtered by an optional `status` parameter.
...
+### `activatevault`
+
+The `activatevault` RPC Command executes the signing process of the
+Revault unvault transaction.
+
+#### Request
+
+| Parameter        | Type    | Description                        |
+| ---------------- | ------- | ---------------------------------- |
+| `txid`           | string  | Unique ID of the vault to activate |
```
[the remark this answers is not in the document]
Comment (darosior) [reply doc/API.md:125]: You missed this one, but this can be added later

File: doc/API.md [hunk 7]
```diff
@@ -77,18 +82,165 @@ filtered by an optional `status` parameter.
...
 
 #### Request
 
-| Parameter        | Type    | Description                         |
-| ---------------- | ------- | ----------------------------------- |
-| `txid`           | string  | Unique ID of the vault to sign      |
-| `only_emergency` | boolean | Sign only the emergency transaction |
+| Parameter        | Type    | Description                                     |
+| ---------------- | ------- | ----------------------------------------------- |
+| `txid`           | string  | Unique ID of the vault to secure                |
```
Comment (darosior) [inline doc/API.md:94]: nit: "Deposit txid"

File: doc/API.md [hunk 8]
```diff
@@ -77,18 +82,165 @@ filtered by an optional `status` parameter.
...
-TODO: specify response
+| Field                  | Type   | Description                                                 |
+| ---------------------- | ------ | ----------------------------------------------------------- |
+| `emergency_tx`         | string | Emergency transaction to sign using the PSBT format         |
+| `cancel_tx`            | string | Cancel transaction to sign using the PSBT format            |
+| `emergency_unvault_tx` | string | Emergency unvault transaction to sign using the PSBT format |
+
+### `revocationtxs`
+
+The PSBTs once signed must be given back to the daemons.
```
Comment (darosior) [inline doc/API.md:106]: Nit: "daemon" (withtout the `s`)

File: doc/API.md [hunk 9]
```diff
@@ -77,18 +82,165 @@ filtered by an optional `status` parameter.
 | ------------- | ------------------------------------------ | ------------------------- |
 | `vaults`      | array of [vault resource](#vault-resource) | Vaults filtered by status |
 
-### `signvault`
+### `getrevocationtxs`
 
-The `signvault` RPC Command executes the signing process of the Revault
-pre-signed transactions.
+The `getrevocationtxs` RPC Command executes the signing process of the
+Revault revocation transactions.
```
Comment (darosior) [inline doc/API.md:88]:
```suggestion
The `getrevocationtxs` RPC Command builds and return the revocation
transactions corresponding to this vault.
```

File: doc/API.md [hunk 10]
```diff
@@ -77,18 +82,165 @@ filtered by an optional `status` parameter.
...
+| Field                  | Type   | Description                                                |
+| ---------------------- | ------ | ---------------------------------------------------------- |
+| `emergency_tx`         | string | Emergency transaction signed using the PSBT format         |
+| `cancel_tx`            | string | Cancel transaction signed using the PSBT format            |
+| `emergency_unvault_tx` | string | Emergency unvault transaction signed using the PSBT format |
+
+### `getunvaulttx`
+
+The `getunvaulttx` RPC Command executes the signing process of the
+Revault unvault transaction.
```
Comment (darosior) [inline doc/API.md:119]:
```suggestion
The `getunvaulttx` RPC Command builds and return the unvault
transaction corresponding to this vault.
```

**Comment by darosior** [comment]
Amended to push the latest changes requested as i'd like to merge this now.

**Comment by darosior** [comment]
ACK d61ad53

Thanks, it'll be really helpful.
I'm going to merge it as is but in the future it'd be great for us to be more rigorous with regard to [the workflow](https://github.com/re-vault/revaultd/blob/master/CONTRIBUTING.md#workflow) (atomic, sane, and meaningful commits) :).

**Pull request merged**